pom.xml

```diff
@@ -47,7 +47,7 @@ under the License.
     </modules>
 
     <properties>
-        <flink.version>1.16.2</flink.version>
+        <flink.version>1.18.0</flink.version>
```

Comment:
Are we going to drop support for flink 1.16, 1.17 together with this commit?

Reply:
Yes, we need the code introduced on 1.18 by this [FLIP-287](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=240880853)

Reply:
I think in that case we probably need to bump connector version as well to something like 4.0.0
WDYT @MartijnVisser ?